lib/src/validator/pubspec_typo.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../entrypoint.dart';
+import '../levenshtein.dart';
+import '../validator.dart';
+
+/// Validates that a package's pubspec does not contain any typos in its keys.
+class PubspecTypoValidator extends Validator {
+  PubspecTypoValidator(Entrypoint entrypoint) : super(entrypoint);
+
+  @override
+  Future validate() async {
+    final fields = entrypoint.root.pubspec.fields;
+
+    if (fields == null) return;
+
+    for (final key in fields.keys) {
+      var bestLevenshteinRatio = 100.0;
+      var closestKey = '';
+
```

Comment:
Should we make a quick check:
`if (_validPubspecKeys.contains(key)) { continue; }`

Just to save some compute time..

Reply:
I guess we're probably not CPU bound, just seems like a really cheap thing to do :D

Reply:
sounds good!